gdc_ng_models/models/misc.py

```diff
@@ -22,7 +23,7 @@ class FileReport(Base):
     node_id = Column('node_id', Text, index=True)
     ip = Column('ip', String)
     country_code = Column('country_code', String, index=True)
-    timestamp = Column('timestamp', DateTime, server_default="now()")
+    timestamp = Column('timestamp', DateTime, server_default=func.now(), primary_key=True)
```

Comment:
I am probably late to this gig, so my comment may have already been consider/discussed. 

1. Is making this column a primary_key required for using it to do partitioning? 
2. Given that partitioning generally partitions rows into nearly equal numbers, and that `timestamp` depends on activity within the given period, which is not guaranteed to be nearly equal or stay nearly equal. Shouldn't the `id` column, which is sequence based, be a good candidate?

Reply:
1. Yes, without making the `timestamp`, the partitioning does not work.
2. I'm not sure if partitioning is strictly meant to divide the table into equal partitions. It's quite commonly used to partition based on dates/timestamps especially in a regime where we want keep the most recent records and archive older ones. Also, this was based on the [design document](https://wiki.uchicago.edu/pages/viewpage.action?spaceKey=CDIS&title=GDC+File+Report+Partitioning) written by @singergr so maybe he can give a better perspective.

Reply:
As Salman mentioned, the goal of the partitioning in this case is to make it possible to age out data gracefully, so we want to partition by timestamp. We are not trying to shard across different DB servers or anything like that.